Recruit-Training/Advanced-Recruit-Training/src/BatteryStateOfChargeService.cpp

```diff
@@ -1,37 +1,107 @@
+#include <math.h>
+#include <QTime>
+
 #include "BatteryStateOfChargeService.h"
+#include "BatteryData.h"
 
 namespace
 {
     const double BATTERY_AMP_HOUR_CAPACITY = 123.0;
+    const int MS_IN_HOURS = 3600000;
+    const int MS_IN_MINUTES = 60000;
+    const int MS_IN_SECONDS = 1000;
 }
 
 BatteryStateOfChargeService::BatteryStateOfChargeService(double initialStateOfChargePercent)
 : initialStateOfChargePercent_(initialStateOfChargePercent)
 {
+    /*
+                : initialStateOfChargePercent_(initialStateOfChargePercent)
+      Same as ^   initialStateOfChargePercent_ = initialStateOfChargePercent;
+      Another example in BatteryData.cpp                                     */
```

Comment:
Get rid of all unnecessary comments, code is usually written to be self-documenting.